src/rust/engine/src/intrinsics.rs

```diff
@@ -298,7 +305,20 @@ fn path_globs_to_digest(
   async move {
     let key = externs::acquire_key_for(args.pop().unwrap())?;
     let digest = context.get(Snapshot(key)).await?;
-    Ok(Snapshot::store_directory(&core, &digest))
+    Ok(Snapshot::store_directory_digest(&core, &digest))
+  }
+  .boxed()
+}
+
+fn path_globs_to_file_listing(
+  context: Context,
+  mut args: Vec<Value>,
+) -> BoxFuture<'static, NodeResult<Value>> {
+  let core = context.core.clone();
+  async move {
+    let key = externs::acquire_key_for(args.pop().unwrap())?;
+    let file_listing = context.get(FileListing(key)).await?;
```

Comment:
One thing to note is that because `path_globs_to_digest` does this work independently (ie, it depends on a different `Node` type which doesn't use the `FileListing` node) some redundant work will happen if someone requests a `FileListing` and then a `Snapshot` for the same `PathGlobs`.

Not a big deal, but worth thinking through what memoized values can/should be reused between the nodes.

Reply:
Should we update `path_globs_to_digest` to call this new `path_globs_to_paths` function? This essentially does this the first have of the digest one.